src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incentives/IncentivesService.kt

```diff
@@ -219,6 +229,153 @@ class IncentivesService(
     )
   }
 
+  @Audit
+  fun createPrisonIncentiveLevelData(
+    prisonId: String,
+    createRequest: CreatePrisonIncentiveRequest,
+  ): PrisonIncentiveLevelDataResponse {
+    val prison = agencyLocationRepository.findByIdOrNull(prisonId)
+      ?: throw BadDataException("Prison with id=$prisonId does not exist")
+
+    val prisonIncentiveLevel =
+      savePrisonIncentiveLevelData(prison, createRequest)
+
+    val visitAllowanceLevel =
+      saveVisitAllowanceLevelData(prison, createRequest)
+
+    telemetryClient.trackEvent(
+      "prison-incentive-level-data-created",
+      mapOf(
+        "code" to createRequest.levelCode,
+        "location" to prisonId,
+        "active" to prisonIncentiveLevel.active.toString(),
+        "defaultFlag" to prisonIncentiveLevel.default.toString(),
+        "remandTransferLimit" to prisonIncentiveLevel.remandTransferLimit.toString(),
+        "remandSpendLimit" to prisonIncentiveLevel.remandSpendLimit.toString(),
+        "convictedTransferLimit" to prisonIncentiveLevel.convictedTransferLimit.toString(),
+        "convictedSpendLimit" to prisonIncentiveLevel.convictedSpendLimit.toString(),
+      ),
+      null,
+    )
+    return PrisonIncentiveLevelDataResponse(
+      prisonId = prisonId,
+      iepLevelCode = createRequest.levelCode,
+      visitOrderAllowance = visitAllowanceLevel.visitOrderAllowance,
+      privilegedVisitOrderAllowance = visitAllowanceLevel.privilegedVisitOrderAllowance,
+      defaultOnAdmission = prisonIncentiveLevel.default,
+      active = prisonIncentiveLevel.active,
+      remandSpendLimitInPence = prisonIncentiveLevel.remandSpendLimit?.let {
+        prisonIncentiveLevel.remandSpendLimit.multiply(
+          BigDecimal.valueOf(100),
+        ).toInt()
+      },
+      remandTransferLimitInPence = prisonIncentiveLevel.remandTransferLimit?.let {
+        prisonIncentiveLevel.remandTransferLimit.times(
+          BigDecimal.valueOf(100),
+        ).toInt()
+      },
+      convictedSpendLimitInPence = prisonIncentiveLevel.convictedSpendLimit?.let {
+        prisonIncentiveLevel.convictedSpendLimit.times(
+          BigDecimal.valueOf(100),
+        ).toInt()
+      },
+      convictedTransferLimitInPence = prisonIncentiveLevel.convictedTransferLimit?.let {
+        prisonIncentiveLevel.convictedTransferLimit.times(
+          BigDecimal.valueOf(100),
+        ).toInt()
```

Comment:
minor but you can do movePointRight(2) instead of multiplying

Reply:
I'll do that and make it an extension function